tests/nnstreamer_filter_tvm/unittest_filter_tvm.cc

```diff
@@ -29,100 +30,145 @@
+    g_autofree gchar *model_name = g_strdup_printf ("tvm_add_one_%s%s_", ARCH, NNSTREAMER_SO_FILE_EXTENSION);
+
+    return g_build_filename (root_path, "tests", "test_models", "models", model_name, NULL);
+  }
 
-/**
- * @brief internal function to get model filename
- */
-static void
-_get_model_file (gchar ** model_file)
-{
-  const gchar *src_root = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
-  gchar *root_path = src_root ? g_strdup (src_root) : g_get_current_dir ();
-  gchar *model_name = g_strdup_printf ("tvm_add_one_%s.so_", ARCH);
-
-  *model_file = g_build_filename (
-      root_path, "tests", "test_models", "models", model_name, NULL);
-  g_free (model_name);
-  g_free (root_path);
-}
+public:
+  /**
+   * @brief Construct a new NNStreamerFilterTVMTest object
+   */
+  NNStreamerFilterTVMTest ()
+    : sp(nullptr), model_file(nullptr), pipeline(nullptr), gstpipe(nullptr), sink_handle(nullptr)
+  {
+  }
 
-/**
- * @brief Signal to validate new output data
- */
-static void
-_check_output (GstElement *element, GstBuffer *buffer, gpointer user_data)
-{
-  GstMemory *mem_res;
-  GstMapInfo info_res;
-  gboolean mapped;
-  gfloat *output;
-  UNUSED (element);
-  UNUSED (user_data);
-
-  mem_res = gst_buffer_peek_memory (buffer, 0);
-  mapped = gst_memory_map (mem_res, &info_res, GST_MAP_READ);
-  ASSERT_TRUE (mapped);
-  output = (gfloat *) info_res.data;
-
-  for (guint i = 0; i < 10; i++) {
-    EXPECT_EQ (1, output[i]);
+  /**
+   * @brief Set tensor filter properties
+   */
+  void SetFilterProperty (GstTensorFilterProperties *prop, const gchar **models)
+  {
+    memset (prop, 0, sizeof (GstTensorFilterProperties));
+    prop->fwname = "tvm";
+    prop->fw_opened = 0;
+    prop->model_files = models;
+    prop->num_models = g_strv_length ((gchar **) models);
   }
 
-  gst_memory_unmap (mem_res, &info_res);
-}
+  /**
+   * @brief SetUp method for each test case
+   */
+  void SetUp () override
+  {
+    wrong_model_files[0] = "temp.so";
+    wrong_model_files[1] = NULL;
+
+    model_file = SetModelFile();
+    proper_model_files[0] = model_file;
+    proper_model_files[1] = NULL;
+
+    input.size = output.size = 0;
+    input.data = nullptr;
+    output.data = nullptr;
+
+    sp = nnstreamer_filter_find ("tvm");
+  }
+
+  /**
+   * @brief TearDown method for each test case
+   */
+  void TearDown () override
+  {
+    g_free (model_file);
+    g_free (pipeline);
+
+    g_clear_object (&gstpipe);
+    g_clear_object (&sink_handle);
+
+    g_free (input.data);
+    g_free (output.data);
+  }
+
+  /**
+   * @brief Signal handler to validate new output data
+   */
+  static void
+  CheckOutput (GstElement *element, GstBuffer *buffer, gpointer user_data)
+  {
+    GstMemory *mem_res;
+    GstMapInfo info_res;
+    gboolean mapped;
+    gfloat *output;
+    UNUSED (element);
+    UNUSED (user_data);
+
+    mem_res = gst_buffer_peek_memory (buffer, 0);
+    mapped = gst_memory_map (mem_res, &info_res, GST_MAP_READ);
```

Comment:
What happens if mem_res is NULL? Returning False or SegFault?

Reply:
If mem_res is NULL, gst_memory_map will return FALSE.